src/main/java/org/opensearch/timeseries/ml/ModelColdStart.java

```diff
@@ -463,8 +463,10 @@ private void getFeatures(
             // make sure the following logic making sense via checking lastRoundFirstStartTime > 0
             if (lastRounddataSample != null && lastRounddataSample.size() > 0) {
                 concatenatedDataSample = new ArrayList<>();
-                concatenatedDataSample.addAll(lastRounddataSample);
+                // since we move farther in history in current one, last round data should come
```

Comment:
Did this bug have significant impact before?

Reply:
yes, if there are sparse data, we will get exception when trying to do two-round sample data collection.